### DISCUSSION TITLE
Libraries updates in .NET 9 Preview 5

### DISCUSSION THREAD

**Discussion by richlander** [dotnet/runtime#103313] [Announcements]
Welcome to .NET 9 Preview 5! Here's what's new in .NET Libraries.

This release contains the following Libraries improvements:
- [Enhanced AI capabilities with `TensorPrimitives` and `Tensor<T>`](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview5/libraries.md#enhanced-ai-capabilities-with-tensorprimitives-and-tensort)
- [`params` offers better performance with `Span` overloads](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview5/libraries.md#params-offers-better-performance-with-span-overloads)
- [`SearchValues` can search for (multiple) substrings within a string](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview5/libraries.md#searchvalues-can-search-for-multiple-substrings-within-a-string)
- [`foreach` completed tasks with `Task.WhenEach`](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview5/libraries.md#foreach-completed-tasks-with-taskwheneach)
- [Prioritized unbounded channel](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview5/libraries.md#prioritized-unbounded-channel)
- [OpenTelemetry activity linking is more flexible](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview5/libraries.md#opentelemetry-activity-linking-is-more-flexible)
- [TypeDescriptor trimming support](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview5/libraries.md#typedescriptor-trimming-support)
- [Type name parsing](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview5/libraries.md#type-name-parsing)

See the [Libraries Release Notes](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview5/libraries.md) for all the details.

.NET 9 Preview 5:

- [Discussion](https://aka.ms/dotnet/9/preview5)
- [Release notes](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview5/README.md)
- [What's new in .NET 9](https://learn.microsoft.com/dotnet/core/whats-new/dotnet-9/overview)

**Comment by wstaelens** [comment]
@richlander  any news on when this one will be accepted for .net 9 ?? https://github.com/dotnet/runtime/pull/97898 ? https://github.com/dotnet/runtime/pull/97898#issuecomment-2046099942 )
(via https://github.com/dotnet/runtime/issues/78629 )

**Comment by jianghupt** [comment]
"Why aren't the runtime and libraries separated?"
